lib/rest-adapter.js

```diff
@@ -403,6 +411,14 @@ RestAdapter.prototype._registerMethodRouteHandlers = function(router,
     // Express 4.x only supports delete
     verb = 'delete';
   }
+
+  //For now: exclude paths with parameters like :id
+  //and handler functions with err obj i.e. (err, req,res,next)
+  if (!route.path.includes(':') && handler.length < 4) {
+    handler.methodName = sharedMethod.name;
+    handler.path = route.fullPath;
+    customRouter.registerPathAndHandlers({ verb: verb, fullPath: route.fullPath }, handler);
+  }
   router[verb](route.path, handler);
```

Comment:
@bajtos : the built-in router places `all` verb handler on the top of the handler stack, and keeps a flag to check if the `path` has an `all` handler, if yes, it invokes the `all` handler first and runs other registered handlers
In our case, `trie` has a placeholder for `all` handler, and we can check, but we are only inovking single handler. 
I'm thinking of changing it in a way that: 
in `all: handler(req,res,next)` , `next` points to the second handler in the stack for the path and so on.. I will commit this approach in the next commit.

Or, should you have any better idea for this scenario -- after looking at recent commit -- please advise. Thank you :)